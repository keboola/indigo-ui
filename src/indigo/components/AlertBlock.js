import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'react-bootstrap';

class AlertBlock extends React.Component {
  render() {
    const { type, title, children } = this.props;
    return (
      <Alert className="alert-block" bsStyle={type}>
        <h3 className="alert-block-title">{title}</h3>
        <div className="alert-block-body">
          <i className="fa fa-exclamation-triangle" />
          {children}
        </div>
      </Alert>
    );
  }
}

AlertBlock.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['warning', 'danger']).isRequired,
  children: PropTypes.any.isRequired,
};

export default AlertBlock;
