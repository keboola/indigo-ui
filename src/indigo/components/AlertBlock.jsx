import React from 'react';
import {
    Alert
} from 'react-bootstrap';

const AlertBlock = React.createClass({
    propTypes: {
        title: React.PropTypes.bool.isRequired,
        style: React.PropTypes.bool.isRequired,
        closable: React.PropTypes.bool,
        children: React.PropTypes.any
    },

    render() {
        return (
            <Alert className="alert-block" bsStyle={this.props.style}>
                <h3 className="alert-block-title">{this.props.title}</h3>
                <div className="alert-block-body">
                    {this.props.children}
                </div>
            </Alert>
        );
    }
});

export default AlertBlock;
