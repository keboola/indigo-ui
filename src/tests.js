import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
