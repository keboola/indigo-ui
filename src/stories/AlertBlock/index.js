import React from 'react';
import {storiesOf} from '@storybook/react';
import {withInfo} from '@storybook/addon-info';
import { Icon } from '../../indigo/components';
import { Row, Col } from 'react-bootstrap';

import AlertBlock from '../../indigo/components/AlertBlock';

let content1 = (
  <ul className="list-unstyled list-no-padding">
    <li>
      <a>
          Keboola Connection Storage (13.24 GB of 10 GB)
      </a>
    </li>
  </ul>
)

let content2 = (
  <ul className="list-unstyled list-no-padding">
    <li>
      <a>
          Keboola Connection Storage (13.24 GB of 10 GB)
      </a>
    </li>
    <li>
      <a>
          Users count (12 of 10)
      </a>
    </li>
  </ul>
)

let content3 = (
  <Row>
    <Col md={6}>
      <h4>
        <Icon.Writer className="icon-category"/>
        <span>Writers</span>
      </h4>
      <ul className="list-unstyled">
        <li>
          <a>
            Custom science PHP
          </a>
        </li>
        <li>
          <a>
            Custom science Python (v2.x)
          </a>
        </li>
      </ul>
      <h4>
        <Icon.Application className="icon-category"/>
        <span>Application</span>
      </h4>
      <ul className="list-unstyled">
        <li>
          <a>
            Gooddata (deprecated)
          </a>
        </li>
        <li>
          <a>
            Pigeon
          </a>
        </li>
        <li>
          <a>
            Pigeon Importer (Deprecated)
          </a>
        </li>
      </ul>
    </Col>
    <Col md={6}>
      <h4>
        <Icon.Extractor className="icon-category"/>
        <span>Extractors</span>
      </h4>
      <ul className="list-unstyled">
        <li>
          <a>
            Gooddata (deprecated)
          </a>
        </li>
        <li>
          <a>
            Pigeon
          </a>
        </li>
        <li>
          <a>
            Pigeon Importer (Deprecated)
          </a>
        </li>
      </ul>
      <h4>
       <Icon.Transformation className="icon-category"/>
        Part 1
      </h4>
      <ul className="list-unstyled">
        <li>
          <a>rwe</a>
        </li>
      </ul>
      <h4>
        <Icon.Transformation className="icon-category"/>
        Part 2
      </h4>
      <ul className="list-unstyled">
        <li>
          <a>sadfas</a>
        </li>
      </ul>
    </Col>
  </Row>
)

let content4 = (
  <Col>
    <Row>
      <Col md={12}>
        <p>Learn more about the deprecation <a>timeline and reasons</a>.</p>
      </Col>
    </Row>
      {content3}
  </Col>
)

let content5 = (
  <Row>
    <Col md={9}>
      <p>
          Migrate your current configurations to new Database Writer. This writer will continue to work until May 2017. Then, all your configurations will be migrated automatically. The migration will also alter your orchestrations to use the new writers. The old configurations will remain intact for now. You can remove them yourself after a successful migration.
      </p>
      <button className="btn btn-primary">
        Proceed to Migration
      </button>
    </Col>
  </Row>
)



storiesOf('AlertBlock', module)
  .add(
    'Styles',
    withInfo({
      text: `
        Warnings & Errors
        `,
      inline: true,
    })(() => {
      return (
        <div>
          <AlertBlock
            type="warning"
            title="Project contains deprecated components">
            {content1}
          </AlertBlock>

          <AlertBlock
            type="danger"
            title="Project is over quota">
            {content1}
          </AlertBlock>
        </div>
      );
    })
  ).add(
  'Recommended contents',
  withInfo({
    text: `
        Here are show receomended usages of content, please respect them. 
        If you are missing other demo content, please get in touch with UX crowd .)
        `,
    inline: true,
  })(() => {
    return (
      <div>
        <AlertBlock
          type="warning"
          title="Alert Block with list content">
          {content2}
        </AlertBlock>

        <AlertBlock
          type="warning"
          title="Alert Block with categorized list content">
          {content3}
        </AlertBlock>

        <AlertBlock
          type="warning"
          title="Alert Block with categorized list content & description">
          {content4}
        </AlertBlock>

        <AlertBlock
          type="warning"
          title="Alert Block with paragraph & action">
          {content5}
        </AlertBlock>
      </div>
    );
  })
);
