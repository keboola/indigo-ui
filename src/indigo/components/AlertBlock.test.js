import React from 'react';

import { snapshot } from '../../tests';
import AlertBlock from './AlertBlock';

describe('<AlertBlock />', () => {
  it('AlertBlock - warning', () => {
    snapshot(
      <AlertBlock title="This is warning" type="warning">
        Warning text
      </AlertBlock>
    );
  });

  it('AlertBlock - danger', () => {
    snapshot(
      <AlertBlock title="This is danger" type="danger">
        Danger text
      </AlertBlock>
    );
  });
});
