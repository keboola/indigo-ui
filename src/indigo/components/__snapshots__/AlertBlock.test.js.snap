// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertBlock /> AlertBlock - danger 1`] = `
<div
  className="alert-block alert alert-danger"
  role="alert"
>
  <h3
    className="alert-block-title"
  >
    This is danger
  </h3>
  <div
    className="alert-block-body"
  >
    <svg
      className="icon alert-block-icon"
      height={19}
      viewBox="0 0 22 19"
      width={22}
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M-1-2h24v24H-1z"
        />
        <path
          d="M0 19h22L11 0 0 19zm12-3h-2v-2h2v2zm0-4h-2V8h2v4z"
          fill="currentColor"
        />
      </g>
    </svg>
    Danger text
  </div>
</div>
`;

exports[`<AlertBlock /> AlertBlock - warning 1`] = `
<div
  className="alert-block alert alert-warning"
  role="alert"
>
  <h3
    className="alert-block-title"
  >
    This is warning
  </h3>
  <div
    className="alert-block-body"
  >
    <svg
      className="icon alert-block-icon"
      height={20}
      viewBox="0 0 20 20"
      width={20}
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M-2-2h24v24H-2z"
        />
        <path
          d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 15H9v-2h2v2zm0-4H9V5h2v6z"
          fill="currentColor"
        />
      </g>
    </svg>
    Warning text
  </div>
</div>
`;
